File: newspark-art/TypeScript/tests/pages/LoginPage.ts
import { Page, Locator } from '@playwright/test';

export class LoginPage {
  private page: Page;
  private username_input: Locator;
  private password_input: Locator;
  private login_button: Locator;
  private logout_button: Locator;
  private url: string;

  constructor(page: Page) {
    this.page = page;
    this.username_input = page.locator('[data-testid="username-input"]');
    this.password_input = page.locator('[data-testid="password-input"]');
    this.login_button = page.locator('[data-testid="login-button"]');
    this.logout_button = page.locator('[data-testid="logout-button"]');
    this.url = "http://localhost:5173/login";
  }

  async navigate(): Promise<void> {
    await this.page.goto(this.url);
  }

  async enterCredentials(username: string, password: string): Promise<void> {
    await this.username_input.fill(username);
    await this.password_input.fill(password);
  }

  async clickLoginButton(): Promise<void> {
    await this.login_button.click();
  }

  async clickLogoutButton(): Promise<void> {
    await this.logout_button.click();
  }

  async isLogoutButtonVisible(): Promise<boolean> {
    await this.logout_button.waitFor({ state: 'visible' });
    return await this.logout_button.isVisible();
  }

  async isUsernameInputVisible(): Promise<boolean> {
    await this.username_input.waitFor({ state: 'visible' });
    return await this.username_input.isVisible();
  }

  async isOnPage(): Promise<boolean> {
    return this.page.url() === this.url;
  }
}

File: newspark-art/Pythonn/pages/login_page.py
from playwright.sync_api import Page

class LoginPage:
    def __init__(self, page: Page):
        self.page = page
        self.username_input = page.locator('[data-testid="username-input"]')
        self.password_input = page.locator('[data-testid="password-input"]')
        self.login_button = page.locator('[data-testid="login-button"]')
        self.logout_button = page.locator('[data-testid="logout-button"]')
        self.url = "http://localhost:5173/login"

    def navigate(self):
        self.page.goto(self.url)

    def enter_credentials(self, username: str, password: str):
        self.username_input.fill(username)
        self.password_input.fill(password)

    def click_login_button(self):
        self.login_button.click()

    def click_logout_button(self):
        self.logout_button.click()

    def is_logout_button_visible(self):
        self.logout_button.wait_for(state='visible')
        return self.logout_button.is_visible()

    def is_on_page(self):
        return self.page.url == self.url
